consul/server_details/server_details_test.go

```diff
@@ -8,6 +8,48 @@ import (
 	"github.com/hashicorp/serf/serf"
 )
 
+func TestServerDetails_Key_params(t *testing.T) {
+	ipv4a := net.ParseIP("127.0.0.1")
+	ipv4b := net.ParseIP("1.2.3.4")
+
+	tests := []struct {
+		name  string
+		sd1   *server_details.ServerDetails
+		sd2   *server_details.ServerDetails
+		equal bool
+	}{
+		{
+			name: "Addr inequality",
```

Comment:
This is actually correct.  The address is different, but the Key comparison succeeds.